assets/src/edit-story/components/fontPicker/scrollList.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback, useRef } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import debounce from '../../utils/debounce';
+
+const List = styled.ul.attrs({ role: 'listbox' })`
+  width: 100%;
+  overflow-x: hidden;
+  overflow-y: auto;
+  overscroll-behavior: none auto;
+`;
+
+const Item = styled.li.attrs({
+  tabIndex: '0',
+  role: 'option',
+})`
+  overflow: hidden;
+`;
+
+function ScrollList({
+  items,
+  itemRenderer,
+  onScroll,
+  currentOffset,
+  className,
+}) {
+  const ref = useRef();
+  const itemOffsets = useRef([]);
+  const numItems = items.length;
+
+  const setOffset = useCallback(
+    (index) => (node) => {
+      if (!node) {
+        return;
+      }
+      itemOffsets.current[index] = node.offsetTop;
+    },
+    []
+  );
+
+  useEffect(() => {
+    const node = ref.current;
+    if (!node) {
+      return undefined;
+    }
+
+    const handleScroll = debounce((evt) => {
```

Comment:
Can't we just use `useDebouncedCallback`?

Reply:
I did check that out, but it does different stuff than just a simple debounced callback.  In this instance I needed the callback inside a `useEffect` to assign as a callback to be deregistered in the same place. For such an occasion it's normally recommended to define the callback locally in `useEffect` - also there's no need for the extra memoization of defining the callback outside the `useEffect`. I can rewrite to that, just to reduce that extra dependency, but not sure it's making the codebase better.

Reply:
Done in 86cd4903.